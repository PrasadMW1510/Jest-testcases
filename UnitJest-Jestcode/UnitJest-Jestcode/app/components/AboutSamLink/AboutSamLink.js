/**
 *
 * AboutSamLink
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import './AboutSamLink.scss';

const AboutSamLink = ({ item, onAboutSAMClick }) => {
  const AboutSAMClick = e => {
    e.preventDefault();
    onAboutSAMClick({ item });
  };

  return (
    <div>
      <button onClick={AboutSAMClick} className="asl-btn-link">
        <span className="asl-btn-span">About SAM </span>
      </button>
    </div>
  );
};

AboutSamLink.propTypes = {
  item: PropTypes.array,
  onAboutSAMClick: PropTypes.func.isRequired,
};

export default AboutSamLink;
