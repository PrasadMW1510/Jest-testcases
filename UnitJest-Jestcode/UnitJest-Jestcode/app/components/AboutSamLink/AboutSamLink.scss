
.asl-btn-span {
  font-family: 'Verdana sans-serif';
  font-size: 1em;
  text-align: left;
  margin: 0;
  outline: none;
  padding: 5px;
  border: 0;
  background-color: transparent;
  cursor: hand;
  color: #00f;
  text-decoration: underline;
}
