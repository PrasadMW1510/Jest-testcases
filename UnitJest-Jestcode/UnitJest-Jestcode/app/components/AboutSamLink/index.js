/**
 *
 * AboutSamLink
 *
 */

export { default } from './AboutSamLink';
