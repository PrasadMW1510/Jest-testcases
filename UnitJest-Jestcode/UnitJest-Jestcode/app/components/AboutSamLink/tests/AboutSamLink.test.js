import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import AboutSamLink from '../index';

describe('<AboutSamLink />', () => {
  let wrapper = null;
  let mockOnAboutClick = null;
  let mockItems = null;
  let mockedItem = null;
  beforeEach(() => {
    mockItems = [
      [{ name: 'SAM Server' }, { name: 'fm' }],
      [{ version: 'v1.1' }, { version: 'v1.2' }],
      [{ Enabled: 'true' }, { Enabled: 'false' }],
      [{ ServerBuildNumber: 'v.3.3' }, { ServerBuildNumber: 'v3.2' }],
      [{ MediaBuildNumber: 'v.3.3' }, { MediaBuildNumber: 'v3.2' }],
      [{ supernumber: 'v.3.3' }, { supernumber: 'v3.2' }],
    ];
    mockedItem = { item: mockItems };
    mockOnAboutClick = jest.fn();
    wrapper = shallow(<AboutSamLink item={mockItems} onAboutSAMClick={mockOnAboutClick} />);
  });

  it('Should render as expected', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('handleOnAboutSamLink', () => {
    it('on click', () => {
      wrapper.find('.asl-btn-link').simulate('click', { preventDefault: () => {} });
      expect(mockOnAboutClick).toBeCalledWith(mockedItem);
    });
  });
});
