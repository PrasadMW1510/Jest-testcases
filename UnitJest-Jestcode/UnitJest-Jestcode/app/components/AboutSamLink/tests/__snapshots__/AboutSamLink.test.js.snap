// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutSamLink /> Should render as expected 1`] = `
<div>
  <button
    className="asl-btn-link"
    onClick={[Function]}
  >
    <span
      className="asl-btn-span"
    >
      About SAM 
    </span>
  </button>
</div>
`;
